Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -245,6 +249,12 @@ - (void)setObject:(id)value forKey:(NSString*)key {
     CFPreferencesSetAppValue(static_cast<CFStringRef>(key), valueCopy, kCFPreferencesCurrentApplication);
 
     [[NSNotificationCenter defaultCenter] postNotificationName:NSUserDefaultsDidChangeNotification object:self];
+
+    if ([_synchronizeQueue operationCount] < 2) {
```

Comment:
Maybe this isn't the best way to determine if an operation should be added to the queue. I'm going to experiment with alternate designs and see if it makes my perf test any faster.